packages/material-ui-lab/src/TreeView/descendants.js

```diff
@@ -16,6 +16,28 @@ function findIndex(array, comp) {
   return -1;
 }
 
+function binaryFindElement(array, element) {
+  let start = 0;
+  let end = array.length - 1;
+
+  while (start <= end) {
+    const middle = Math.floor((start + end) / 2);
+
+    if (array[middle].element === element) {
+      return middle;
+    }
+    // Is before
+    // eslint-disable-next-line no-bitwise
+    if (array[middle].element.compareDocumentPosition(element) & Node.DOCUMENT_POSITION_PRECEDING) {
```

Comment:
This exists :open_mouth: Is it an issue if elements are portaled?

Reply:
Hmm, It shouldn't be but I should add a test :P